test/object_store/test_backup.py

```diff
@@ -106,6 +106,37 @@ async def test_backup_to_non_existent_bucket(manager: ManagerClient, s3_server):
     assert status['state'] == 'failed'
 
 
+@pytest.mark.asyncio
+async def test_backup_to_non_existent_snapshot(manager: ManagerClient, s3_server):
+    '''backup should fail if the snapshot does not exist'''
+
+    cfg = {'enable_user_defined_functions': False,
+           'object_storage_config_file': str(s3_server.config_file),
+           'experimental_features': ['keyspace-storage-options'],
+           'task_ttl_in_seconds': 300
+           }
+    cmd = ['--logger-log-level', 'snapshots=trace:task_manager=trace']
+    server = await manager.server_add(config=cfg, cmdline=cmd)
+    ks, cf = await prepare_snapshot_for_backup(manager, server)
+
+    prefix = f'{cf}/backup'
+    tid = await manager.api.backup(server.ip_addr, ks, cf, 'nonexistent-snapshot',
```

Comment:
Isn't it possible for this to fail, like I mentioned here https://github.com/scylladb/scylladb/issues/21381#issuecomment-2447769047 ?
This exception I mean
https://github.com/scylladb/scylladb/blob/64122b3df3bdc729258a644ffbce388fe0732813/db/snapshot/backup_task.cc#L52-L54
Personally, in unit tests it did not replicate in 1000 tries, but if its possible with nodetool it should be possible here too.

Reply:
With ttl set to 300, I don't understand why it doesn't fail here

Reply:
> Isn't it possible for this to fail, like I mentioned here [#21381 (comment)](https://github.com/scylladb/scylladb/issues/21381#issuecomment-2447769047) ? This exception I mean
> 
> https://github.com/scylladb/scylladb/blob/64122b3df3bdc729258a644ffbce388fe0732813/db/snapshot/backup_task.cc#L52-L54
> 

as explained in https://github.com/scylladb/scylladb/issues/21381#issuecomment-2449048132 . we validate the inputs when performing the task, not when launching it.

> Personally, in unit tests it did not replicate in 1000 tries, but if its possible with nodetool it should be possible here too.

Reply:
> With ttl set to 300, I don't understand why it doesn't fail here

is your concern related to #21425 ?